src/components/interpolatedColorLegend.ts

```diff
@@ -109,6 +123,30 @@ export module Component {
       }
     }
 
+    /**
+     * Apply a gradient to the InterpolatedColorLegend.
+     *
+     * @param {boolean} isGradient Whether there should be a gradient or not.
+     *
+     * @returns {InterpolatedColorLegend} The calling InterpolatedColorLegend.
+     */
+     public gradient(isGradient: boolean): InterpolatedColorLegend {
+       this._isGradient = isGradient;
+       return this;
+     }
+
+     /**
+      * Expand the InterpolatedColorLegend to the full length/height of the chart.
+      *
+      * @param {boolean} isExpanded Whether the legend is expanded or not.
+      *
+      * @returns {InterpolatedColorLegend} The calling InterpolatedColorLegend.
+      */
+    public expand(isExpanded: boolean): InterpolatedColorLegend {
+      this._isExpanded = isExpanded;
+      return this;
+    }
```

Comment:
Eventually we will want to expose the `_numSwatches` private variable. Because swatches are fixed height/width for a given component configuration, a user could keep adding/removing swatches until they reach their desired length. Does this sound like a reasonable way to approach the custom sizing issue?